version: '3.7'
services:
    mysql:
        image: mysql:8.0
        restart: on-failure
        environment:
            MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
            MYSQL_DATABASE: ${MYSQL_DATABASE}
        ports:
            - "${MYSQL_PORT}:3306"

    php:
        build:
            context: .
            dockerfile: docker/php/Dockerfile
        restart: on-failure
        volumes:
            - symfony-docker-app-sync:/var/www/html:nocopy
        environment:
            PHP_IDE_CONFIG: ${PHP_IDE_CONFIG}
            TZ: Europe/Warsaw
            XDEBUG_CONFIG: "remote_host=docker.for.mac.localhost"
        external_links:
            - mysql
        depends_on:
            - mysql
        extra_hosts:
            - "local_host:${LOCAL_ADDR_IP}"
        links:
            - mysql

    nginx:
        image: nginx:1.15.3-alpine
        restart: on-failure
        volumes:
            - ./docker/nginx/default.conf:/etc/nginx/conf.d/default.conf:ro
            - symfony-docker-app-sync:/var/www/html:nocopy
        ports:
            - "${NGINX_PORT}:80"
        depends_on:
            - php
    adminer:
        image: adminer
        restart: on-failure
        ports:
            - "${ADMINER_PORT}:8080"

volumes:
    symfony-docker-app-sync:
        external: true